src/Polysemy/Law.hs

```diff
@@ -0,0 +1,131 @@
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE UndecidableInstances   #-}
+{-# LANGUAGE ViewPatterns           #-}
+
+module Polysemy.Law
+  ( Law (..)
+  , runLaw
+  , MakeLaw (..)
+  , Citizen (..)
+  , module Test.QuickCheck
+  ) where
+
+import Control.Arrow (first)
+import Data.Char
+import Polysemy
+import Test.QuickCheck
+
+
+class Citizen r a | r -> a where
```

Comment:
We'll probably need to have some comment here, or maybe some different name - personally I had to read through the code to get the idea of what `Citizen` means :slightly_smiling_face: